jib-core/scripts/prepare_release.sh

```diff
@@ -4,39 +4,43 @@
 set -e
 
 Colorize() {
-	echo "$(tput setff $2)$1$(tput sgr0)"
+  echo "$(tput setff $2)$1$(tput sgr0)"
 }
 
 EchoRed() {
-	echo "$(tput setaf 1; tput bold)$1$(tput sgr0)"
+  echo "$(tput setaf 1; tput bold)$1$(tput sgr0)"
 }
 EchoGreen() {
-	echo "$(tput setaf 2; tput bold)$1$(tput sgr0)"
+  echo "$(tput setaf 2; tput bold)$1$(tput sgr0)"
 }
 
 Die() {
-	EchoRed "$1"
-	exit 1
+  EchoRed "$1"
+  exit 1
 }
 
 DieUsage() {
-    Die "Usage: ./scripts/prepare_release.sh <release version>"
+  Die "Usage: ./prepare_release.sh <release version> [<post-release-version>]"
 }
 
 # Usage: CheckVersion <version>
 CheckVersion() {
-    [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+$ ]] || Die "Version not in ###.###.### format."
+  [[ $1 =~ ^[0-9]+\.[0-9]+\.[0-9]+(-[0-9A-Za-z]+)?$ ]] || Die "Version: $1 not in ###.###.###[-XXX] format."
 }
 
-[ $# -ne 2 ] || DieUsage
+[ $# -ne 2 ] || [ $# -ne 3 ] || DieUsage
```

Comment:
Is this right? I think the original code was also wrong, if the intention was to die if the number of args isn't 2. It should have been `[ $# -ne 2] && DieUsage` or `[ $# -eq 2 ] || DieUsage`.

Reply:
Ah yes this logic is just wrong.